### ISSUE TITLE
Git pulls are blocking

### ISSUE THREAD

**Issue by SamLau95** [data-8/interact#6]
Sometimes a `git pull` takes a long time.

This is on `data8.b.e`:

```
[03/11/2016 10:11:03 PM]: INFO -- Starting new HTTPS connection (1): data8.berkeley.edu
[03/11/2016 10:11:03 PM]: DEBUG -- "GET /hub/api/users/samlau95 HTTP/1.1" 200 127
[03/11/2016 10:11:03 PM]: INFO -- Starting pull.
[03/11/2016 10:11:03 PM]: INFO --     User: samlau95
[03/11/2016 10:11:03 PM]: INFO --     Repo: data8assets
[03/11/2016 10:11:03 PM]: INFO --     Paths: ['labs/project1']
[03/11/2016 10:11:03 PM]: INFO -- Existing paths in sparse-checkout: ['labs/lab01', 'labs/lab01/lab01.ipynb', 'labs/lab02', 'labs/lab03', 'mentors', 'labs/lab05']
[03/11/2016 10:11:03 PM]: INFO -- ['labs/project1'] written to sparse-checkout
[03/11/2016 10:11:06 PM]: INFO -- Starting pull from origin

vvvvv this is 5 minutes later vvvvv
[03/11/2016 10:17:58 PM]: INFO -- Pulled from origin
[03/11/2016 10:17:58 PM]: INFO -- Redirecting to /user/samlau95/tree/data8assets/labs/project1
[03/11/2016 10:17:58 PM]: INFO -- /home/samlau95/data8assets chown'd to samlau95
```

While this pull was happening, I was not able to request more pulls and it doesn't look like others were able to either.

I suspect this happened because I'm pulling for the first time in a couple weeks. Still, it would be better if pulls could happen asynchronously.

**Comment by ryanlovett** [comment]
Apparently flask [doesn't support](http://flask.pocoo.org/docs/0.10/design/) async operations. It might be necessary to replace it with something like tornado.

Once that happens, the hub needs to tell the user that the git pull is in progress which means updating the UI and communicating with the thread doing the git pull.

**Comment by papajohn** [comment]
Solved by #8
